### PR TITLE
Reduce udf processor

### PR DESCRIPTION
**Pull request by yhl25**

### CODE REVIEW

**Comment by whynowy** [comment]
@yhl25 - I had some changes on my local didn't get a chance to commit (just did it), would you like to do it based on the latest?

**Comment by yhl25** [comment]
> @yhl25 - I had some changes on my local didn't get a chance to commit (just did it), would you like to do it based on the latest?
yeah saw the latest. I had the same changes :)

**Comment by whynowy** [comment]
Some of the changes were copied from yours, just wanted to make it complete, so that you can take over to continue.

**Comment by yhl25** [comment]
Okayy..

File: pkg/reduce/options.go [hunk 1]
```diff
@@ -11,6 +11,8 @@ type Options struct {
 	readBatchSize int64
 	// windowOpts Options for window
 	windowOpts *window.Options
+	// vertexType indicates the type of the vertex
+	vertexType dfv1.VertexType
```
Comment (whynowy) [inline pkg/reduce/options.go]: Is this needed? The code for reduce is only supposed to be used for reduce vertex.

File: pkg/udf/common.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		}
+	}
+
+	return reader, writers
+}
+
+func buildJetStreamBufferIO(ctx context.Context, fromBufferName string, vertexInstance *dfv1.VertexInstance) (isb.BufferReader, map[string]isb.BufferWriter, error) {
+	writers := make(map[string]isb.BufferWriter)
+
+	fromStreamName := fmt.Sprintf("%s-%s", vertexInstance.Vertex.Spec.PipelineName, fromBufferName)
```
Comment (whynowy) [inline pkg/udf/common.go]: Use isbsvc.JetStreamName()

File: pkg/reduce/options.go (hunk 1 above)
Comment (yhl25) [reply pkg/reduce/options.go]: yeah, I don't think we need that. removed it.

File: pkg/udf/common.go (hunk 2 above)
Comment (yhl25) [reply pkg/udf/common.go]: done

**Pull request merged**